src/modules/InstURL.rb

```diff
@@ -142,232 +83,45 @@ def RewriteCDUrl(url)
-          url = Ops.add(
-            Ops.add(Ops.add(url, option_separator), "proxy="),
-            proxy
-          )
-          option_separator = "&"
-        end
-        proxyport = Linuxrc.InstallInf("ProxyPort")
-        if proxyport != nil && proxyport != ""
-          url = Ops.add(
-            Ops.add(Ops.add(url, option_separator), "proxyport="),
-            proxyport
-          )
-          option_separator = "&"
-        end
-        proxyproto = Linuxrc.InstallInf("ProxyProto")
-        if proxyproto != nil && proxyproto != ""
-          url = Ops.add(
-            Ops.add(Ops.add(url, option_separator), "proxyproto="),
-            proxyproto
-          )
-          option_separator = "&"
-        end
-        proxyuser = Linuxrc.InstallInf("ProxyUser")
-        if proxyuser != nil && proxyuser != ""
-          url = Ops.add(
-            Ops.add(Ops.add(url, option_separator), "proxyuser="),
-            proxyuser
-          )
-          option_separator = "&"
-        end
-        proxypassword = Linuxrc.InstallInf("ProxyPassword")
-        if proxypassword != nil && proxypassword != ""
-          url = Ops.add(
-            Ops.add(Ops.add(url, option_separator), "proxypassword="),
-            proxypassword
-          )
-          option_separator = "&"
-        end
-        workgroup = Linuxrc.InstallInf("WorkDomain")
-        if workgroup != nil && workgroup != ""
-          url = Ops.add(
-            Ops.add(Ops.add(url, option_separator), "workgroup="),
-            workgroup
-          )
-          option_separator = "&"
-        end
-
-        if instmode == "https"
-          Builtins.y2milestone("HTTPS instmode detected")
-
-          if !SSLVerificationEnabled()
-            Builtins.y2security(
-              "Disabling certificate check for the installation repository"
-            )
-
-            # libzypp uses ssl_verify=no option
-            url = Ops.add(Ops.add(url, option_separator), "ssl_verify=no")
-            option_separator = "&"
-          end
-        end
-      end # is_network
-
-      if options != ""
-        url = Ops.add(Ops.add(url, option_separator), options)
-        option_separator = "&"
-        Builtins.y2milestone("options %1", options)
-      end
+      @installInf2Url = URL.Build(parts)
+      log.info "Using install URL: #{URL.HidePassword(@installInf2Url)}"
+      @installInf2Url
+    end
 
-      url = Builtins.sformat("iso:/?iso=%1&url=%2", isoimg, url) if isoimg != ""
+    # Schemes considered local for installInf2Url
+    LOCAL_SCHEMES = ["cd", "dvd", "hd"]
 
-      Builtins.y2debug("URL %1", URL.HidePassword(url))
-      url
+    # Determines whether the installation URL is remote or not
+    #
+    # @return [Boolean] true if it's remote; false otherwise.
+    # @see installInf2Url
+    def is_network
+      return @is_network unless @is_network.nil?
+      scheme = URL.Parse(installInf2Url("")).fetch("scheme")
+      @is_network = !LOCAL_SCHEMES.include?(scheme)
```

Comment:
NTH: I'd use `scheme.downcase` here, just to be more error proof when somebody writes `INSTALL=HTTP://EXAMPLE.COM`...